drivers/power_domain/power_domain_psl.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2022 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_npcx_power_domain_psl
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/pm/device.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(power_domain_psl, CONFIG_POWER_DOMAIN_LOG_LEVEL);
+
+struct npcx_psl_config {
+	struct gpio_dt_spec enable;
+	/* PSL_IN pinmux configuration */
+	const struct pinctrl_dev_config *pcfg;
+};
+
+static int npcx_psl_pm_action(const struct device *dev,
+			     enum pm_device_action action)
+{
+	const struct npcx_psl_config *config = dev->config;
+	int ret = 0;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_TURN_OFF:
+		/* Configure detection settings of PSL_IN pads first */
+		ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_SLEEP);
+		if (ret < 0) {
+			LOG_ERR("PSL_IN pinctrl setup failed (%d)", ret);
+			return ret;
+		}
+
+		/* Turn the Core Domain power supply (VCC1) off */
+		gpio_pin_set_dt(&config->enable, 1);
```

Comment:
I'm confused by this.  The "level" mode for the PSL output describes setting PSL_OUT/GPIO85 high (active) to enable VCC1, and setting PSL_OUT/GPIO85 low (inactive) to turn off VCC1.

Your code above seems to be doing the opposite.

Reply:
@fabiobaltieri asked a similar question before. [Here ](https://github.com/zephyrproject-rtos/zephyr/pull/45931#discussion_r881233161)is my reply in the previous discussion. But I found a typo in this comment. 
> GPIO85 is not a standard GPIO and PSL_OUT is switched to ```inactive``` state only by setting bit 7 in PDOUT reg of gpio8 module. 

It should be bit 5, not 7. This information is from the paragraph, ```A transition from 0 to 1 of GPIO85 data-out bit 5 in P8DOUT register indicates a firmware request to dessert ...```, of npcx datasheet section 4.8.1.

Reply:
Thanks, please add a comment to the code describes this non-standard behavior of GPIOP85.

Reply:
Done.